src/window.cpp

```diff
@@ -771,19 +771,27 @@ void WindowManager::AddToStack(Window *w)
 	stack->lower = w;
 }
 
+/**
+ * Called by a window prior to its deletion.
+ * @param w Window being deleted.
+ */
+void WindowManager::PreRemove(Window *w)
+{
+	if (this->current_window) this->current_window = nullptr;
```

Comment:
Shouldn't it compare with `w`?
That was what the previous version did I think.

I am also pondering on the name `PreDelete` is perhaps better, or `Unmanage` :p  
I can live with `PreRemove` too though, any of those names seems fine.